test/reference-implementations.test.ts

```diff
@@ -33,29 +36,32 @@ describe('reference implementation tests', () => {
             getBIP44CoinTypeToAddressPathTuple({ address_index: index }),
           );
 
-          expect(await childNode.getAddress()).toStrictEqual(expectedAddress);
+          expect(childNode.address).toStrictEqual(expectedAddress);
         }
       });
     });
 
     describe('deriveKeyFromPath', () => {
       it('derives the expected keys', async () => {
         // Ethereum coin type key
-        const parentKey = await deriveKeyFromPath([
+        const [parentKey, , parentChainCode] = await deriveKeyFromPath([
           mnemonicBip39Node,
           BIP44PurposeNodeToken,
           `bip32:60'`,
         ]);
 
         for (let index = 0; index < addresses.length; index++) {
           const expectedAddress = addresses[index];
-          const childKey = await deriveKeyFromPath(
+          const [childKey] = await deriveKeyFromPath(
             getBIP44CoinTypeToAddressPathTuple({ address_index: index }),
             parentKey,
+            undefined,
+            parentChainCode,
           );
 
+          const publicKey = secp256k1.getPublicKey(childKey, false);
           expect(
-            privateKeyToEthAddress(childKey).toString('hex'),
+            publicKeyToEthAddress(publicKey).toString('hex'),
```

Comment:
Should we check that we can derive correctly from both priv and pub or is that overkill?

Reply:
The public key is directly derived from the private key, in the same way that the `privateKeyToEthAddress` function would do it, so I don't think it's necessary?

Reply:
Up to you!